notify.go

```diff
@@ -0,0 +1,39 @@
+package autonat
+
+import (
+	"time"
+
+	inet "github.com/libp2p/go-libp2p-net"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+var _ inet.Notifiee = (*AmbientAutoNAT)(nil)
+
+func (as *AmbientAutoNAT) Listen(net inet.Network, a ma.Multiaddr)      {}
+func (as *AmbientAutoNAT) ListenClose(net inet.Network, a ma.Multiaddr) {}
+func (as *AmbientAutoNAT) OpenedStream(net inet.Network, s inet.Stream) {}
+func (as *AmbientAutoNAT) ClosedStream(net inet.Network, s inet.Stream) {}
+
+func (as *AmbientAutoNAT) Connected(net inet.Network, c inet.Conn) {
+	p := c.RemotePeer()
+
+	go func() {
+		// add some delay for identify
+		time.Sleep(250 * time.Millisecond)
```

Comment:
Why can't we just walk over each connected peer and check lazily? That way, we don't even *need* this check.

Reply:
I don't think we gain anything by the lazy check. 
We'll have to iterate through the list of peers, which also adds complexity for recognizing new autonat servers after we have iterated once.
So we might as well do it at connect time, the goroutine cost is marginal.

Reply:
I'm not worried about the goroutine cost, I'm worried about the race. However, this is probably fine as an initial implementation.

Reply:
The race is probably benign, as we are just collecting peer IDs of peers that implement the protocol.

Reply:
Yeah, I'm just worried that if the peer supported AutoNAT –but their Identify took longer than 250ms– we will have missed that peer forever, no?

Reply:
We can increase a tad... but how much is too much?